File: javascript/matmult.js
// Matrix multiplication in javascript

SIZE = 512

a = new Array(SIZE)
c = new Array(SIZE)

function matmult() {
  for (var i = 0; i < SIZE; ++i) {
    a[i] = new Array(SIZE)
    c[i] = new Array(SIZE)
  }

  for (var i = 0; i < SIZE; ++i)
  for (var j = 0; j < SIZE; ++j) {
    a[i][j] = 1
    c[i][j] = 0
  }

  for (var i = 0; i < SIZE; ++i)
    for (var k = 0; k < SIZE; ++k)
      for (var j = 0; j < SIZE; ++j)
        c[i][j] += a[i][j] * a[i][k]

  res = 0
  for (var i = 0; i < SIZE; ++i)
    for (var j = 0; j < SIZE; ++j)
      res += c[i][j];

  console.log (res)
}

matmult()

File: perl/matmult.pl
# Matrix multiplication in perl

$SIZE = 512;

my @a;
my @b;

sub matmult {

	for $i (0..$SIZE-1) {
		for $j (0..$SIZE-1) {
			$a[i][j] = 1;
			$c[i][j] = 0;
		}
  }

	for $i (0..$SIZE-1) {
		for $k (0..$SIZE-1) {
			for $j (0..$SIZE-1) {
				$c[i][j] += $a[i][k] * $a[k][j];
			}
		}
	}

	$res = 0;
	for $i (0..$SIZE-1) {
		for $j (0..$SIZE-1) {
			$res += $c[i][j];
		}
	}

	print "$res";
}

matmult();